internal/blob/downloader.go

```diff
@@ -25,18 +27,25 @@ const (
 )
 
 type downloader struct {
-	logger       logging.Logger
-	baseCacheDir string
+	logger           logging.Logger
+	baseCacheDir     string
+	blobOptions      []Option
+	validationSha256 string
 }
 
-func NewDownloader(logger logging.Logger, baseCacheDir string) *downloader { //nolint:golint,gosimple
-	return &downloader{
+func NewDownloader(logger logging.Logger, baseCacheDir string) downloader { //nolint:golint,gosimple
+	return downloader{
 		logger:       logger,
 		baseCacheDir: baseCacheDir,
 	}
 }
 
-func (d *downloader) Download(ctx context.Context, pathOrURI string) (Blob, error) {
+type DownloadOption func(*downloader)
+
+func (d downloader) Download(ctx context.Context, pathOrURI string, options ...DownloadOption) (Blob, error) {
```

Comment:
🤔 I'm really curious why we're removing the pointer receiver, through this whole file.

Reply:
Ya this is so that we can set `downloader.blobOptions` & `downloader.validationSha256` in the download method & prevent it mutating the underlying `downloader` object

Reply:
This sounds like a smell. What is the purpose of associating `blobOptions` to the downloader when it's really only an option for a single operation? Same can be said for `validationSha256`. It looks like it's adding unnecessary state to a service (`Downloader`) which now isn't reusable because of its state....

I would propose looking for alternatives to providing optional arguments.

For example, `DownloadOptions` can be a `func (*downloadOptions)` and `downloadOptions` be an unexported struct with the two fields you added to `Downloader`. 

Then instead of

```
	for _, option := range options {
		option(&d)
	}
```

do something like:

```
        do := &downloadOptions{}
	for _, option := range options {
		option(&do)
	}
```